ccswbs/gus
# Redirects

# Summary of changes
- Installed gatsby-plugin-client-side-redirect
- Created redirect on Gatsby (client side) using Drupal redirects

## Frontend
- Redirects should now exist (based on Drupal redirects)

## Backend
**Multidev:** [https://redirects-chug.pantheonsite.io](https://redirects-chug.pantheonsite.io)
- Enabled Redirect module and Redirect 404 module
- Allow Publishers, Gatsby Build and Gatsby Preview to administer individual redirects
- Unauthenticated users do not have access to redirect JSON data

# Test Plan
Note: If you are testing locally, you will need to run `gatsby build` rather than `gatsby develop`, then navigate to the public folder to find the page with the client-side redirect on it. Alternatively, you can test via Gatsby Cloud.

- Visit [https://build-59002b1c-66be-4309-be74-990a12d8c831.gtsb.io/testing-redirects](https://build-59002b1c-66be-4309-be74-990a12d8c831.gtsb.io/testing-redirects) which should redirect to another page
- Go to [https://redirects-chug.pantheonsite.io](https://redirects-chug.pantheonsite.io)
- Add a new basic page or program
- Either compile locally or use Gatsby Cloud (connecting your site to the redirects branch using the redirects-chug.pantheonsite.io) to create a new build 
- View the page you created on the compiled build (copy the URL)
- Back on Drupal, change the name of the page or program (this should create a new redirect)
- Compile again locally (run gatsby build instead of Gatsby develop if running things locally) or trigger a new build on your connected Gatsby Cloud site
- Once the build is done, go back to the URL you copied. It should redirect to the updated URL.
- If compiling locally, go into the public directory that was generated by gatsby build and navigate to the page you previously created (a.k.a. same URL as the page you copied). It should have a client-side redirect on the page.

# Caveat
I believe this will only work for Content Hub pages that are already available via the gateway. (So if there's a blanket rule where all pages under /my-department/ are available, then if you change the name of /my-department/my-page, it will redirect to the new page as long as it also lives under /my-department/)
